src/Discord.Net.WebSocket/ConnectionManager.cs

```diff
@@ -103,12 +103,10 @@ public virtual async Task StartAsync()
                 finally { _stateLock.Release(); }
             });
         }
-        public virtual async Task StopAsync()
+        public virtual Task StopAsync()
         {
             Cancel();
-            var task = _task;
```

Comment:
What happens to `_task` now?

Reply:
It'll complete as normal without blocking. Refer to https://github.com/RogueException/Discord.Net/pull/1095#issue-196925154 for better explanation.

Reply:
Gotcha.